### PR TITLE
add cell footprint support & add it to equiv check

### PR DESCRIPTION
**Pull request by AcKoucher**
This is to address some [Resizing Problems](https://github.com/The-OpenROAD-Project/OpenROAD/issues/5577) in OR. With this, RSZ won't swap cells that are functionally equivalent but non equivalent footprint wise.

### CODE REVIEW

**Comment by jjcherry56** [comment]
If you have an int -> string map there is no point in having the vector.
A better solution would be a set of strings and skip all the extraneous indexing code.
But I really don't see any point in trying to save a few strings the footprints.
The bulk of the memory in a liberty file is the tables, not something like a footprint
And if you do go to the trouble of using indices as a way to make unique strings, there is no point
in using strcmp to compare them when an integer compare would suffice.

OpenSTA does not use OpenROAD formatting for code and unfortunately clang-format does not work well enough to do it for you.
See OpenSTA/doc/CodingGuidelines or look at the other code in the file to see what the standard is.

**Comment by AcKoucher** [comment]
@jjcherry56 Thanks for the feedback. I simplified it by just keeping the string within the cell and adapted the curly brackets to OpenSTA formatting. Please, let me know if there's more to address.

File: liberty/EquivCells.cc [hunk 1]
```diff
@@ -525,4 +526,11 @@ equivCellTimingArcSets(const LibertyCell *cell1,
   }
 }
 
+bool
+equivCellFootprints(const LibertyCell *cell1,
+                    const LibertyCell *cell2)
+{
+  return strcmp(cell1->footprint(), cell2->footprint()) == 0;
```
Comment (jjcherry56) [inline liberty/EquivCells.cc:533]: stringEq

File: liberty/Liberty.cc [hunk 2]
```diff
@@ -1989,6 +1992,21 @@ LibertyCell::ensureVoltageWaveforms(const DcalcAnalysisPtSeq &dcalc_aps)
...
 
+void
+LibertyCell::setFootprint(const char *footprint)
+{
+  footprint_ = stringCopy(footprint);
+}
+
+const char*
+LibertyCell::footprint() const
+{
```
Comment (jjcherry56) [inline liberty/Liberty.cc:2003]: This should just return nullptr if there is no footprint instead of ""

**Comment by AcKoucher** [comment]
Review addressed.

**Pull request merged**